Mathlib/CategoryTheory/Generator.lean

```diff
@@ -567,4 +569,98 @@ theorem wellPowered_of_isDetector [HasPullbacks C] (G : C) (hG : IsDetector G) :
   haveI := small_subsingleton ({G} : Set C)
   wellPowered_of_isDetecting hG
 
+theorem wellPowered_of_isSeparator [HasPullbacks C] [Balanced C] (G : C) (hG : IsSeparator G) :
+    WellPowered C := wellPowered_of_isDetecting hG.isDetector
+
+section HasGenerator
+
+variable (E : Type u₃) [Category.{v₃} E]
+
+/--
+A category instantiates `HasSeparator` if and only if it has a separator.
+See also `IsSeparator`.
+-/
+class HasSeparator : Prop :=
+  hasSeparator : ∃ G : E, IsSeparator G
+
```

Comment:
I think it might be nice to add
```lean
variable (C)
noncomputable def separator [HasSeparator C] : C :=
  Classical.indefiniteDescription _ (HasSeparator.hasSeparator) |>.1

theorem isSeparator_separator [HasSeparator C] : IsSeparator (separator C) :=
  Classical.indefiniteDescription _ (HasSeparator.hasSeparator) |>.2
```
for people who just want to quickly get hold of a separator using choice (similar to how `limit` is for `HasLimit`).